### PR TITLE
Make kube.Client able to talk to remote clusters.

### PR DESCRIPTION
**Pull request by spxtr**
This introduces a config struct containing the remote cluster info. It is mostly based off of the struct that you get when you do `gcloud container clusters describe`, except most fields are removed.

### CODE REVIEW

File: prow/kube/client.go [hunk 1]
```diff
@@ -195,6 +198,72 @@ func NewClientInCluster(namespace string) (*Client, error) {
...
+	}
+	var c Cluster
+	if err := json.Unmarshal(data, &c); err != nil {
+		return nil, err
+	}
+	return NewClient(&c, namespace)
+}
+
+// NewClient returns an authenticated Client using the keys in the Cluster.
+func NewClient(c *Cluster, namespace string) (*Client, error) {
```
Comment (rmmh) [inline prow/kube/client.go]: this should probably have a test with some dummy data, to prevent regressions

Comment (spxtr) [inline prow/kube/client.go]: It's certainly possible but it's a lot of work grumble grumble also a test server won't be identical to a k8s api server grumble grumble fine.

Comment (spxtr) [inline prow/kube/client.go]: Done.

**Comment by rmmh** [comment]
/lgtm

thanks for the test!

**Pull request merged**